nixos/bootspec: adopt the merged RFC-0125 (#224489)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2305.section.md

- `libxcrypt`, the library providing the `crypt(3)` password hashing function, is now built without support for algorithms not flagged [`strong`](https://github.com/besser82/libxcrypt/blob/v4.4.33/lib/hashes.conf#L48). This affects the availability of password hashing algorithms used for system login (`login(1)`, `passwd(1)`), but also Apache2 Basic-Auth, Samba, OpenLDAP, Dovecot, and [many other packages](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20libxcrypt&type=code).

> - [RFC-0125](https://github.com/NixOS/rfcs/pull/125) has been merged; [`boot.bootspec.enable`](options.html#opt-boot.bootspec.enable) is now on by default, this means you will have a bootspec document called `boot.json` generated for each system and specialisation in the top-level, e.g. `/run/current-system`. This is useful to enable advanced boot usecases in NixOS such as SecureBoot or to cleanup our bootloader story.

Reviewer: ```suggestion
- [`boot.bootspec.enable`](options.html#opt-boot.bootspec.enable) is now enabled by default because [RFC-0125](https://github.com/NixOS/rfcs/pull/125) was merged. This means you will have a bootspec document called `boot.json` generated for each system and specialisation in the top-level. This is useful to enable advanced boot usecases in NixOS such as SecureBoot.
```

The option is more valuable that the RFC for the release notes, IMO, so put that first.

Author: agreed
